loky/backend/_thread_pool_limiters.py

```diff
@@ -0,0 +1,506 @@
+    os.environ.setdefault("KMP_DUPLICATE_LIB_OK", "True")
+
+
+# Structure to cast the info on dynamically loaded library. See
+# https://linux.die.net/man/3/dl_iterate_phdr for more details.
+UINT_SYSTEM = ctypes.c_uint64 if sys.maxsize > 2**32 else ctypes.c_uint32
+UINT_HALF_SYSTEM = ctypes.c_uint32 if sys.maxsize > 2**32 else ctypes.c_uint16
+
+
+class dl_phdr_info(ctypes.Structure):
+    _fields_ = [
+        ("dlpi_addr",  UINT_SYSTEM),       # Base address of object
+        ("dlpi_name",  ctypes.c_char_p),   # path to the library
+        ("dlpi_phdr",  ctypes.c_void_p),   # pointer on dlpi_headers
+        ("dlpi_phnum",  UINT_HALF_SYSTEM)  # number of element in dlpi_phdr
+        ]
+
+
+# List of the supported implementations. The items hold the prefix of loaded
+# shared objects, the name of the internal_api to call, matching the
+# MAP_API_TO_FUNC keys and the name of the user_api, in {"blas", "openmp"}.
+SUPPORTED_IMPLEMENTATIONS = [
+    {
+        "filename_prefixes": ("libiomp", "libgomp", "libomp", "vcomp",),
+        "internal_api": "openmp",
+        "user_api": "openmp"
+    },
+    {
+        "filename_prefixes": ("libopenblas",),
+        "internal_api": "openblas",
+        "user_api": "blas"
+    },
+    {
+        "filename_prefixes": ("libmkl_rt", "mkl_rt",),
+        "internal_api": "mkl",
+        "user_api": "blas"
+    },
+]
+
+# map a internal_api (openmp, openblas, mkl) to set and get functions
+MAP_API_TO_FUNC = {
+    "openmp": {
+        "set": "omp_set_num_threads",
+        "get": "omp_get_max_threads"},
+    "openblas": {
+        "set": "openblas_set_num_threads",
+        "get": "openblas_get_num_threads"},
+    "mkl": {
+        "set": "MKL_Set_Num_Threads",
+        "get": "MKL_Get_Max_Threads"},
+}
+
+# Helpers for the doc and test names
+ALL_USER_APIS = set(impl['user_api'] for impl in SUPPORTED_IMPLEMENTATIONS)
+ALL_PREFIXES = [prefix for impl in SUPPORTED_IMPLEMENTATIONS
+                for prefix in impl['filename_prefixes']]
+INTERN_APIS = list(MAP_API_TO_FUNC.keys())
+
+
+class _ThreadPoolLibrariesWrapper:
+    # Wrapper around classic C-libraries for scientific computations which use
+    # thread-pools. Give access to functions to set and get the maximum number
+    # of threads they are allowed to used for inner parallelism.
+
+    cls_thread_locals = threading.local()
+
+    def __init__(self):
+        self._load()
+
+    def get_limit(self, prefix, user_api, limits):
+        if prefix in limits:
+            return limits[prefix]
+        if user_api in limits:
+            return limits[user_api]
+        return None
+
+    def set_thread_limits(self, limits=None, user_api=None):
+        """Limit maximal number of threads used by supported C-libraries.
+
+        Return a list with the modules where the maximal number of threads
+        available has been modified.
+        """
+        if isinstance(limits, int) or limits is None:
+            if user_api in ('all', None):
```

Comment:
Actually, we could get rid of `"all"` as the placeholder for no filtering. I think `user_api=None` is good enough for that purpose. WDYT?